### PR TITLE
Wip/hughsie/require plugin hints

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: plugins/dell/dell.quirk [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
+UefiVersionFormat = quad
 
-[fwupd-daemon-version-format]
-com.dell.uefi*.firmware=none
+# DEPRECATED: put this in the AppStream metadata itself using
+# <custom>
+#   <value key="LVFS::VersionFormat">quad</value>
+# </custom>
+[DaemonVersionFormat=Quad]
+ComponentIDs = com.dell.uefi*.firmware
```
Comment (superm1) [inline plugins/dell/dell.quirk]: Do you want to still see this land in the CAB file itself in the future or let LVFS do it on the OEM account? I would prefer to keep it in the OEM account.

File: src/fu-engine.c [hunk 2]
```diff
@@ -971,12 +971,19 @@ fu_engine_vendor_quirk_release_version (FuEngine *self, AsApp *app)
...
 
-	/* any quirks match */
-	quirk = fu_quirks_lookup_by_glob (self->quirks,
-					  FU_QUIRKS_DAEMON_VERSION_FORMAT,
-					  as_app_get_id (app));
-	if (g_strcmp0 (quirk, "none") == 0)
-		flags = AS_VERSION_PARSE_FLAG_NONE;
+	/* fall back to the quirk database until all files have metadata */
+	quirk = fu_quirks_lookup_by_id (self->quirks,
+					"DaemonVersionFormat=Quad",
```
Comment (superm1) [inline src/fu-engine.c]: I think you should remain consistent in case, i've seen both Quad and quad.

File: data/tests/quirks.d/merged.quirk [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[USB\VID_0A5C&PID_6412]
+Flags = MERGE_ME
```
Comment (superm1) [inline data/tests/quirks.d/merged.quirk]: Do you really want to be using `MERGE_ME`?  I would think using one of the existing flags like `is-bootloader` should be better.

File: src/fu-engine.c (hunk 2 above)
Comment (hughsie) [reply src/fu-engine.c]: Agreed, fixed.

File: plugins/dell/dell.quirk (hunk 1 above)
Comment (hughsie) [reply plugins/dell/dell.quirk]: I don't think it matters as long as something is setting it in the metadata tag. I'm happy for this quirk to live here for a long time, but I don't really want to see any more vendors adding to the list.

File: data/tests/quirks.d/merged.quirk (hunk 3 above)
Comment (hughsie) [reply data/tests/quirks.d/merged.quirk]: That's kinda the point, it's a nonstandard flag. I'll add some more work on this to make it all a bit clearer.

**Comment by hughsie** [comment]
I've pushed this with some changes, thanks for the review. As a side-effect, fwupd starts up so much quicker when you have an absurd number of USB devices connected.

**Comment by hughsie** [comment]
> Did you also push 83d46b0 at the same time?

Well, it was in a different branch -- but we need to do it as meson is going to turn this warning into an error soon. Can you just revert it as a local patch in the 18.04 package?

**Comment by superm1** [comment]
Yeah I can, but also it's going to make it harder for people to test master on 18.04 too (unpackaged build).  I've got an idea, i'll send a separate PR.

**Pull request closed without merging**